modules/dnn/src/layers/pooling_layer.cpp

```diff
@@ -504,10 +505,11 @@ class PoolingLayerImpl : public PoolingLayer
                     {
                         int xstart = x0 * stride_w - pad_w;
                         int xend = min(xstart + kernel_w, inp_width + pad_w);
-                        int xdelta = xend - xstart;
                         xstart = max(xstart, 0);
                         xend = min(xend, inp_width);
-                        float inv_kernel_area = 1.f/(ydelta*xdelta);
+                        float inv_kernel_area =
+                            padMode == "SAME" ? (yend - ystart) * (xend - xstart) : kernel.area();
```

Comment:
> padMode == "SAME"

Please avoid strings check inside of for loops.
Please declare `bool isSamePad` variable outsize of these loops.

Reply:
Thanks! Fixed.